app/src/main/java/org/smartregister/fhircore/PatientListViewModel.kt

```diff
@@ -69,11 +70,13 @@ class PatientListViewModel(application: Application, private val fhirEngine: Fhi
         return liveObservations
     }
 
-    private fun getSearchResults(): List<Patient> {
-        val searchResults: List<Patient> = fhirEngine.search()
+    private fun getSearchResults(filterCriterion: FilterCriterion? = null): List<Patient> {
+        val searchBuilder = fhirEngine.search()
             .of(Patient::class.java)
-            //.filter(string(Patient.ADDRESS_CITY, ParamPrefixEnum.EQUAL, "NAIROBI"))
-            .run()
+
+        if(filterCriterion != null) searchBuilder.filter(filterCriterion)
```

Comment:
Not an expert here but can we use`filterCriterion?.let` ?

Reply:
@vincent-karuri Made a few improvements regarding that